File: app/src/main/java/com/foxek/simpletimer/ui/interval/dialog/WorkoutEditDialog.java
package com.foxek.simpletimer.ui.interval.dialog;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.EditText;
import android.widget.TextView;

import com.foxek.simpletimer.R;
import com.foxek.simpletimer.di.component.ActivityComponent;
import com.foxek.simpletimer.ui.base.BaseFragment;
import com.foxek.simpletimer.ui.interval.IntervalContact;

import javax.inject.Inject;

import androidx.annotation.NonNull;
import androidx.fragment.app.DialogFragment;

import butterknife.BindView;
import butterknife.ButterKnife;
import butterknife.OnClick;
import butterknife.Unbinder;

import static com.foxek.simpletimer.utils.Constants.EXTRA_WORKOUT_NAME;

public class WorkoutEditDialog extends BaseFragment {

    @Inject
    IntervalContact.Presenter presenter;

    private Unbinder binder;

    @BindView(R.id.training_edit_text)
    EditText workoutEditText;

    public static WorkoutEditDialog newInstance(String workoutName) {
        WorkoutEditDialog mWorkoutEditDialog = new WorkoutEditDialog();
        Bundle args = new Bundle();
        args.putString(EXTRA_WORKOUT_NAME, workoutName);
        mWorkoutEditDialog.setArguments(args);
        return mWorkoutEditDialog;
    }

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setStyle(DialogFragment.STYLE_NORMAL, R.style.CustomDialog);
    }

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        View dialogView = inflater.inflate(R.layout.dialog_edit_workout, container, false);

        ActivityComponent component = getActivityComponent();

        if (component != null) {
            component.inject(this);
            binder = ButterKnife.bind(this, dialogView);
        }

        workoutEditText.setText(getArguments().getString(EXTRA_WORKOUT_NAME));
        workoutEditText.setSelection(workoutEditText.getText().length());

        getDialog().setCanceledOnTouchOutside(true);

        return dialogView;
    }

    @OnClick(R.id.delete_button)
    void onDeleteButtonClick() {
        if (getShowsDialog()) {
            workoutEditText.setCursorVisible(false);
            getDialog().cancel();
            presenter.deleteWorkoutButtonClicked();
        } else
            dismiss();
    }

    @OnClick(R.id.save_button)
    void onSaveButtonClick() {
        if (!workoutEditText.getText().toString().isEmpty()) {
            workoutEditText.setCursorVisible(false);
            dismiss();
            presenter.saveWorkoutButtonClicked(workoutEditText.getText().toString());
        } else
            dismiss();
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
        workoutEditText.setCursorVisible(false);
        binder.unbind();
    }
}

File: app/src/main/java/com/foxek/simpletimer/ui/interval/dialog/WorkoutEditDialog.kt
package com.foxek.simpletimer.ui.interval.dialog

import android.os.Bundle
import android.view.View

import com.foxek.simpletimer.R
import com.foxek.simpletimer.ui.base.BaseDialog
import com.foxek.simpletimer.ui.interval.IntervalContact

import javax.inject.Inject

import com.foxek.simpletimer.utils.Constants.EXTRA_WORKOUT_NAME
import kotlinx.android.synthetic.main.dialog_edit_workout.*

class WorkoutEditDialog : BaseDialog() {

    @Inject
    lateinit var presenter: IntervalContact.Presenter

    override var dialogTitle = R.string.dialog_training_setting_title

    override var dialogDescription = R.string.dialog_training_setting_description

    override fun getLayout() = R.layout.dialog_edit_workout

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        activityComponent?.inject(this)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        dialog_edit_workout_field_name.apply {
            setText(arguments?.getString(EXTRA_WORKOUT_NAME))
            setSelection(dialog_edit_workout_field_name.text.length)
        }

        dialog_edit_workout_delete_btn.setOnClickListener {
            dismiss()
            if (showsDialog)
                presenter.deleteWorkoutButtonClicked()
        }

        dialog_edit_workout_save_btn.setOnClickListener {
            dismiss()
            if (checkNotEmpty(dialog_edit_workout_field_name))
                presenter.saveWorkoutButtonClicked(dialog_edit_workout_field_name.text.toString())
        }
    }

    override fun onDestroyView() {
        super.onDestroyView()
        dialog_edit_workout_field_name.isCursorVisible = false
    }

    companion object {
        fun newInstance(name: String?): WorkoutEditDialog {
            val mWorkoutEditDialog = WorkoutEditDialog()
            mWorkoutEditDialog.arguments = Bundle().apply {
                putString(EXTRA_WORKOUT_NAME, name)
            }
            return mWorkoutEditDialog
        }
    }
}